lab_4_summarization_textrank/main.py

```diff
@@ -3,14 +3,54 @@
 Summarize text using TextRank algorithm
 """
 from typing import Union
+import re
+from itertools import chain, combinations
 
 from lab_3_keywords_textrank.main import TextEncoder, \
-    TextPreprocessor
+    TextPreprocessor, TFIDFAdapter
 
 PreprocessedSentence = tuple[str, ...]
 EncodedSentence = tuple[int, ...]
 
 
+class NoRelevantTextsError(Exception):
+    """
+    Raised if there are 0 relevant texts.
+    """
+
+
+class IncorrectQueryError(Exception):
+    """
+    Raised if the query is incorrect.
+    """
+
+
+def arg_check(*args: tuple) -> bool:
+    """
+    Accepts tuples with objects and expected types.
+    Raises a ValueError if any object is empty when it should not be or has the wrong type.
+    Or if arguments are not tuples. Or tuples are too short.
+    Returns True if everything is okay.
+    Positions in tuples:
+    0 = data
```

Comment:
Why can't you set 4 arguments instead of 1? It will be absolutely clear.

Reply:
Because I have already done such functions in the previous labs and I wanted to get to another level, to call a function once to check all arguments no matter how many of them are there instead of just calling it multiple times